### PR TITLE
Fix some light-emitting blocks letting light through (MC-119932)

### PR DESCRIPTION
**Pull request by bs2609**
Fixes vanilla bug [MC-119932](https://bugs.mojang.com/browse/MC-119932).

See `World.getRawLight` for context.

Vanilla's code here treats opaque blocks that emit light as if the opacity was 1.
This may cause incorrect lighting for those blocks, due to the light level of neighbouring blocks being taken into account and retransmitted.

The fix here is to keep the correct opacity value, instead changing the handling of opaque blocks to return the block's own light value instead of 0.

As a small bonus, this also removes the need for a call to `getLightValue` for skylight checks.

### CODE REVIEW

File: patches/minecraft/net/minecraft/world/World.java.patch [hunk 1]
```diff
@@ -715,22 +715,30 @@
...
 -            int j2 = p_175638_2_ == EnumSkyBlock.SKY ? 0 : iblockstate1.func_185906_d();
 -            int k2 = iblockstate1.func_185891_c();
-+            int blockLight = iblockstate1.func_177230_c().getLightValue(iblockstate1, this, p_175638_1_);
-+            int j2 = p_175638_2_ == EnumSkyBlock.SKY ? 0 : blockLight;
++            int j2 = p_175638_2_ == EnumSkyBlock.SKY ? 0 : iblockstate1.func_177230_c().getLightValue(iblockstate1, this, p_175638_1_);
 +            int k2 = iblockstate1.func_177230_c().getLightOpacity(iblockstate1, this, p_175638_1_);
  
 -            if (k2 >= 15 && iblockstate1.func_185906_d() > 0)
-+            if (k2 >= 15 && blockLight > 0)
++            if (k2 >= 15 && false) // Forge: fix MC-119932
```
Comment (liach) [inline patches/minecraft/net/minecraft/world/World.java.patch]: You can replace the whole thing with `if (false)`

**Review by liach — APPROVED** [review]
:+1:

**Comment by bs2609** [comment]
Any update on this?

**Comment by mezz** [comment]
I'm usually pretty reluctant to mess with lighting but this one seems pretty straightforward and I didn't find any obvious issues in testing.

**Comment by LexManos** [comment]
This needs updating, and screenshots showing what is actually changed.

**Comment by bs2609** [comment]
Vanilla:


Forge (patched):

**Comment by LexManos** [comment]
Im not seeing why this is a needed change? We'll get complaints from user saying that we've screwed up the lighting in their base. You're making glowstone dimmer in that screenshot... that's not good.

**Comment by bs2609** [comment]
The visual difference in the screenshot is caused by the differing skylight levels, the light level emitted by glowstone is unchanged.

**Pull request merged**